### PR TITLE
Add simulated network encryption API

### PR DESCRIPTION
**Pull request by ryanorendorff**

### CODE REVIEW

**Comment by ryanorendorff** [comment]
Through this we learned that the deterministic encryption produces some binary blob that differs at least between macos and linux. It seems to be consistent on those platforms though. It is out of scope to determine what the OS may do the cause the error.

File: src/pack.rs [hunk 1]
```diff
@@ -40,7 +40,14 @@ impl FHESerialize for PublicKey {
     where
         Self: Sized,
     {
-        deserialize(bytes).map_err(|_| FheError::InvalidEncoding)
+        let result = deserialize(bytes).map_err(|_| FheError::InvalidEncoding);
+
+        if let Err(ref e) = result {
+            // This is a hack to support the old serialization format.
+            println!("Failed to deserialize the public key");
+        }
```
Comment (samtay) [inline src/pack.rs:48]: What's going on here? We print the deserialization error, but it doesn't look like any kind of hack or support for older formats.

**Review by samtay — APPROVED** [review]
We should remove that `println!` before merging, otherwise good to go.

File: src/pack.rs (hunk 1 above)
Comment (ryanorendorff) [reply src/pack.rs:48]: Oh crap sorry this should not have been committed. I was debugging because solidity doesn't show anything.

**Pull request merged**